### PR TITLE
[Impeller] Support YUV external textures on iOS

### PR DESCRIPTION
**Pull request by ColdPaleLight**
fix https://github.com/flutter/flutter/issues/113688

Test result
2UMlNL5Lm8

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by ColdPaleLight** [comment]
@chinmaygarde @bdero The PR is ready for review :)

File: impeller/renderer/backend/gles/texture_gles.cc [hunk 1]
```diff
@@ -99,6 +99,16 @@ struct TexImage2DData {
         external_format = GL_ALPHA;
         type = GL_UNSIGNED_BYTE;
         break;
+      case PixelFormat::kR8UNormInt:
+        internal_format = GL_RED;
+        external_format = GL_RED;
+        type = GL_UNSIGNED_BYTE;
+        break;
+      case PixelFormat::kR8G8UNormInt:
+        internal_format = GL_RG;
```
Comment (chinmaygarde) [inline impeller/renderer/backend/gles/texture_gles.cc:108]: I don't believe these are part of OpenGL ES 2.0 are they?

**Review by chinmaygarde — COMMENTED** [review]
Apologies for the delay in this review. Everything looks great except the concern that the OpenGL ES implementation doesn't use ES 2.0 Core features. I haven't explored how to reconcile this limitation ATM. If we can't do this ATM, I'd rather leave the implementation out of this attempt and we can revisit it later during an ES backend polishing effort. Thanks!

File: impeller/renderer/backend/gles/texture_gles.cc (hunk 1 above)
Comment (ColdPaleLight) [reply impeller/renderer/backend/gles/texture_gles.cc]: Good catch! I removed these and add a TODO comment with issue id in the test.

**Review by chinmaygarde — APPROVED** [review]

**Pull request merged**